scripts/extract_documentation.py

```diff
@@ -0,0 +1,56 @@
+import csv
+import typing as t
+from pathlib import Path
+from bs4 import BeautifulSoup
+import re
+
+NEWLINE = r'\n'
+assert len(NEWLINE) == 2
+
+FIELD = 'text'
+FIELDNAMES = (FIELD,)
+CSV_CFG = dict(delimiter=' ', quotechar='|', quoting=csv.QUOTE_MINIMAL)
+TAGS = 'p', 'h1', 'h2', 'h3', 'h4', 'h5',
+
+
+def categorize(f: Path):
+    parents = [p.name for p in f.parents]
+    if not any(p.startswith('.') or p.startswith('_') for p in parents):
+        if parents[0] == 'source':
+            return 'source'
+        else:
+            return 'docs'
```

Comment:
What does this function categorize? I can't quite get it from the source code.

Reply:
It's almost a placeholder, it has three outcomes.

If it returns `None`, it means "ignore this file".  if it returns `source` it means it's API documentation, otherwise it's "documentation".

The "docs" and "source" extractions go into different files `docs.csv` and `source.csv` so we could experiment.

Probably in a production version we'd only generate a single file, and then have `--include` and `--exclude` flags to include or exclude what you wanted, and if you wanted two different files you'd run it twice with two different arguments...